### DISCUSSION TITLE
Can't authenticate to REST API if App Service Authentication is enabled

### DISCUSSION THREAD

**Discussion by andysbolton** [polymind-inc/acmebot#594] [Q&A] [answered]
Hi, first of all thanks for the great product. 

I notice if I enable [App Service authentication](https://github.com/shibayan/keyvault-acmebot/wiki/Getting-Started#4-enable-app-service-authentication), I can view the dashboard but receive a 401 when I make a `POST` request to `/api/certificate` with my host key. If I turn off App Service authentication, I am able to make the `POST` request.

Are these authentication methods mutually exclusive or is there another step I'm missing?

**Comment by n7on** [comment]
I'm having same problem. Would really like to know if there are any solution to this.

**Comment by shibayan** [comment, marked as answer]
Yes, when App Service Authentication is enabled, the REST API cannot be accessed with just a host key; to access the REST API with App Service Authentication enabled, you must use Azure AD's Service Principal must be created and an access token obtained.

**Reply by scastellanosnqn** [reply]
shibayan how do you use the token on the REST API ? There are some special headers ? Or just replace the function_host_key ?

**Comment by scastellanosnqn** [comment]
> Hi, first of all thanks for the great product.
> 
> I notice if I enable [App Service authentication](https://github.com/shibayan/keyvault-acmebot/wiki/Getting-Started#4-enable-app-service-authentication), I can view the dashboard but receive a 401 when I make a `POST` request to `/api/certificate` with my host key. If I turn off App Service authentication, I am able to make the `POST` request.
> 
> Are these authentication methods mutually exclusive or is there another step I'm missing?

Andy, have you managed to use the REST API with App Service authentication token ? Could you elaborate ?
